pkg/component/component.go

```diff
@@ -372,17 +421,43 @@ func PushLocal(client *occlient.Client, componentName string, applicationName st
 	}
 	targetPath := fmt.Sprintf("%s/src", s2iSrcPath)
 
+	// If there are files identified as deleted, propagate them to the component pod
+	if len(delFiles) > 0 {
+		glog.V(4).Infof("propogating deletion of files %s to pod", strings.Join(delFiles, " "))
+		/*
+			Delete files observed by watch to have been deleted from each of s2i directories like:
+				deployment dir: In interpreted runtimes like python, source is copied over to deployment dir so delete needs to happen here as well
+				destination dir: This is the directory where s2i expects source to be copied for it be built and deployed
+				working dir: Directory where, sources are copied over from deployment dir from where the s2i builds and deploys source.
+							 Deletes need to happen here as well otherwise, even if the latest source is copied over, the stale source files remain
+				source backup dir: Directory used for backing up source across multiple iterations of push and watch in component container
+								   In case of python, s2i image moves sources from destination dir to workingdir which means sources are deleted from destination dir
+								   So, during the subsequent watch pushing new diff to component pod, the source as a whole doesn't exist at destination dir and hence needs
+								   to be backed up.
+		*/
+		err := client.PropagateDeletes(pod.Name, delFiles, getS2IPaths(pod.Spec.Containers[0].Env))
+		if err != nil {
+			return errors.Wrapf(err, "unable to propagate file deletions %+v", delFiles)
+		}
+	}
+
 	// Copy the files to the pod
 
-	s := log.Spinner("Copying files to pod")
-	err = client.CopyFile(path, pod.Name, targetPath, files)
-	if err != nil {
-		s.End(false)
-		return errors.Wrap(err, "unable push files to pod")
+	if !isForcePush {
+		if len(files) == 0 && len(delFiles) == 0 {
+			return fmt.Errorf("pass files modifications/deletions to sync to component pod or force push")
+		}
+	}
+
+	if isForcePush || len(files) > 0 {
+		glog.V(4).Infof("Copying files %s to pod", strings.Join(files, " "))
```

Comment:
why was this degraded to debug message?
The user should see this message. The only issue with it is that it should say "component" instead of "pod"

Reply:
Shall I make it as Copying changes to component ? because it now includes even delete?

Reply:
Or you can say, "Pushing files/changes to component"

Reply:
I think that copying is fine.

Reply:
Ack I thought it shouldn't be there...
Restoring it back